pkg/network/setup/network.go

```diff
@@ -136,19 +136,28 @@ func preConfigStateRun(nics []podNIC) ([]podNIC, error) {
 
 func discoverPodInterfaces(nics []podNIC) ([]podNIC, error) {
 	for idx, nic := range nics {
-		ifaceLink, err := link.DiscoverByNetwork(nic.handler, nic.vmi.Spec.Networks, *nic.vmiSpecNetwork)
+		podIfaceName, err := discoverPodInterfaceName(nic.handler, nic.vmi.Spec.Networks, *nic.vmiSpecNetwork, *nic.vmiSpecIface)
 		if err != nil {
-			if nic.vmiSpecIface.State != v1.InterfaceStateAbsent {
-				return nil, err
-			}
-			// For a network marked for removal, it is reasonable not to find any interface.
-		} else {
-			nics[idx].podInterfaceName = ifaceLink.Attrs().Name
+			return nil, err
 		}
+		nics[idx].podInterfaceName = podIfaceName
```

Comment:
This assignment changes the input slice in place.
I find this behavior to be unexpected because the function returns a `[]podNIC`.
Please consider creating a new slice and return it.

Reply:
The was the previous behavior as well (`nics[idx].podInterfaceName = ifaceLink.Attrs().Name`), this PR doesn't change it.
It can be fixed as part of a follow-up, I [refer not to include it in this big enough PR.